### DISCUSSION TITLE
Nuxt should provide flexibility in naming files.

### DISCUSSION THREAD

**Discussion by yogeshgalav** [nuxt/nuxt#19800] [Ideas]
As Nuxt implements folder based routing developers could only name file as 
index, create, edit, etc.

What if it reads filename only after underscore so names like
post_index, post_create, post_edit, etc.
so developers could get more flexibility and better experience.

**Comment by danielroe** [comment]
That's not really an option because developers might want underscores in their paths. But you can provide full configuration of the inferred route pattern with the `path` option in `definePageMeta`, which will override what your file name is.

**Reply by yogeshgalav** [reply]
@danielroe 
underscore in url?
It's a totally bad practice, isn't it?
If nuxt is promoting better folder structure then it should promote better url structure too.

**Comment by Blechlawine** [comment]
Another possible option could be similar to what Solid Start allows to fix the problem of having multiple `index.ts` files in your codebase: 
```
posts
    (post).ts
pages
    (pages).ts
```
this would be the same as:
```
posts
    index.ts
pages
    index.ts
```

This could be adapted to also work with parts of filenames:
```
posts
    (post)create.ts
    (post)edit.ts
pages
    (page)create.ts
    (page)edit.ts
```
In this case the stuff in parentheses just gets omitted from the route resulting in these routes:
- posts/create
- posts/edit
- pages/create
- pages/edit
